dask/dataframe/io/tests/test_sql.py

```diff
@@ -154,6 +154,12 @@ def test_npartitions(db):
     assert data.npartitions == 1
     assert (data.name.compute() == df.name).all()
 
+    data_1 = read_sql_table('test', db, columns=list(df.columns),
+                          bytes_per_chunk=2**30,
+                          index_col='number', head_rows=10)
+    assert data_1.npartitions == 1
```

Comment:
ok
working on it

Reply:
commited

Reply:
Currently, there are only 427 bytes in the first 5 rows, 86 in the very first, so a `bytes_per_chunk` of 173 would show a difference between the old, incorrect, behaviour and now.

Reply:
Total size of 7 record : 603
Prev chunk_size : 120.6 (head_rows = 5)
Cur chunk_size : 86.14 (head_rows = 7)
For head_rows = 1, size is 166 which is higher. I think we should pass value  greater than 5 for accurate value

Reply:
I agree, that in general more rows is a good thing here.

I mean just for the test: in the old code, `head_rows=1` would have calculated a per-row size of 427, but with the change, you will get the correct value of 86. So if you also pass `bytes_per_chunk=250`, you should get two partitions (we round down) instead of 1, that is the test.

Reply:
ok got it